### PR TITLE
[WIP] Implement histogram calculation

### PR DESCRIPTION
**Pull request by IshitaTakeshi**
I just implemented the minimum function of histogram calculation for GPU.
I'd also like to determine the interface and write tests if required.

### CODE REVIEW

File: cupy/statistics/histogram.py [hunk 1]
```diff
@@ -3,7 +3,42 @@
...
+    Returns:
+        cupy.ndarray: hist
+        The values of the histogram.
+        cupy.ndarray: bin_edges
+        The bin edges
+
+    .. seealso:: :func:`numpy.histogram`
+    """
+
+    y = cupy.zeros(bins.size-1, dtype=cupy.int32)
```
Comment (unnonouno) [inline cupy/statistics/histogram.py]: Insert spaces before and after `-`.

File: cupy/statistics/histogram.py [hunk 2]
```diff
@@ -3,7 +3,42 @@
...
+    cupy.ElementwiseKernel(
+        'S x, raw T bins, int32 n_bins',
+        'raw int32 y',
+        """
+        int high = n_bins-1;
```
Comment (unnonouno) [inline cupy/statistics/histogram.py]: Insert spaces.

File: cupy/statistics/histogram.py [hunk 3]
```diff
@@ -3,7 +3,42 @@
...
+        int low = 0;
+
+        while(high-low > 1) {
```
Comment (unnonouno) [inline cupy/statistics/histogram.py]: Insert a space after `while`. Insert spaces before and after `-`.

File: cupy/statistics/histogram.py [hunk 4]
```diff
@@ -3,7 +3,42 @@
...
+            int mid = (int)(low + (high-low) / 2);
```
Comment (unnonouno) [inline cupy/statistics/histogram.py]: Insert spaces.

File: cupy/statistics/histogram.py [hunk 5]
```diff
@@ -3,7 +3,42 @@
...
+            if(bins[mid] <= x) {
```
Comment (unnonouno) [inline cupy/statistics/histogram.py]: Insert a space.

**Comment by IshitaTakeshi** [comment]
I think I can

**Comment by IshitaTakeshi** [comment]
I'm trying to write tests regarding the ones of 'bincount' but what they do is not clear for me

**Comment by unnonouno** [comment]
I think `histgram` has the same interface with numpy's one. So, all you need to do is to check both implementations return the same results. 
`for_all_dtypes` decorator calls the function many times with all `dtype`s.
`numpy_cupy_allclose` decorator also calls the function with both `xp=cupy` and `xp=numpy`, and check if two results have same values.

**Comment by unnonouno** [comment]
Can i take over this PR?

**Comment by IshitaTakeshi** [comment]
@unnonouno yes please

**Comment by IshitaTakeshi** [comment]
Thanks a lot!

**Pull request merged**